src/StoreApi/Utilities/InvalidCartException.php

```diff
@@ -0,0 +1,74 @@
+<?php
+namespace Automattic\WooCommerce\Blocks\StoreApi\Utilities;
+
+use WP_Error;
+
+/**
+ * InvalidCartException class.
+ *
+ * @internal This exception is thrown if the cart is in an erroneous state.
+ */
+class InvalidCartException extends \Exception {
+	/**
+	 * Sanitized error code.
+	 *
+	 * @var string
+	 */
+	public $error_code;
+
+	/**
+	 * Additional error data.
+	 *
+	 * @var array
+	 */
+	public $additional_data = [];
+
+	/**
+	 * All errors to display to the user.
+	 *
+	 * @var WP_Error
+	 */
+	public $error;
+
+	/**
+	 * Setup exception.
+	 *
+	 * @param string   $error_code      Machine-readable error code, e.g `woocommerce_invalid_product_id`.
+	 * @param WP_Error $error           The WP_Error object containing all errors relating to stock availability.
+	 * @param array    $additional_data Extra data (key value pairs) to expose in the error response.
+	 */
+	public function __construct( $error_code, $error, $additional_data = [] ) {
```

Comment:
We can add type hinting here if the `$error` must be a `WP_Error` object.

Reply:
Done!